### DISCUSSION TITLE
Feature Idea For Scheduled Cron Jobs

### DISCUSSION THREAD

**Discussion by dchapman1988** [sidekiq/sidekiq#5628] [Ideas]
Just wanted to plant a seed here. It would be really nice to have some UI control over scheduled cron jobs. One feature I often find myself wanting using Sidekiq is the ability to start a scheduled job "now" and skip its next scheduled run time. One could also imagine how totally awesome it would be to expand UI control over scheduled cron jobs, for instance, the actual schedule would be amazing to change on the fly without the need for a deployment.

**Comment by mperham** [comment]
I've wanted to run a cron job occasionally but I usually wanted to tweak something or other about it, like change an argument. I think the Rails console is still the best place for arbitrary out of schedule job running.

Your app's "official" cron schedules are configured in Ruby code, in your initializer. I don't see any way to keep them in sync if editing in the Web UI.

**Comment by dannnylo** [comment]
I created a gem that adds a button to UI that executes a periodic job and you could pause periodic job executions.

![periodic_run_index](https://github.com/dannnylo/sidekiq-belt/assets/20794/0cc900bc-6925-4139-affd-f41b81318727)
![periodic_run_show](https://github.com/dannnylo/sidekiq-belt/assets/20794/086be190-af8e-44d9-bbf6-eed94b7314a0)

![periodic_pause](https://github.com/dannnylo/sidekiq-belt/assets/20794/41fbcee4-9c5b-45cd-b6f7-c359a22f3979)
![periodic_unpause](https://github.com/dannnylo/sidekiq-belt/assets/20794/ea06ae37-068e-4f66-ab10-d83970545a59)

https://rubygems.org/gems/sidekiq-belt
https://github.com/dannnylo/sidekiq-belt
